### PR TITLE
Use common  HTTP Client on .NET Fx.

### LINKED ISSUE #3546: [Bug] Provide better HttpClient factory on .NET fwk (use the factory from .NET Core)
Looks like the .NET fwk factory is throwing a lot of timeout errors: 

image

Suggested fix: use .NET Core's factory on .net

### PR DESCRIPTION
**Pull request by pmaytak**
Fixes #3546

**Changes proposed in this request**
- On .NET Fx targets, use the common HTTP Client which is used on NET Core, (which has better performance and is more resilient).

**Testing**
Existing tests. Tested manually.

**Performance impact**
See related issue. The client should throw less exceptions.

**Documentation**
N/A

### CODE REVIEW

**Review by trwalke — APPROVED** [review]
So this will use the default abstract virtual method which initializes the simple cleint factory right? LGTM

File: src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopPlatformProxy.cs [hunk 1]
```diff
@@ -245,11 +244,6 @@ public override IDeviceAuthManager CreateDeviceAuthManager()
             return new NetDesktopDeviceAuthManager();
         }
 
-        public override IMsalHttpClientFactory CreateDefaultHttpClientFactory()
-        {
-            return new NetDesktopHttpClientFactory();
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopPlatformProxy.cs:250]: Are the associated `NetDesktopHttpClientFactory` and dependencies classes used anywhere? If not, please delete the dead code.

**Review by bgavrilMS — APPROVED** [review]

**Comment by pmaytak** [comment]
> So this will use the default abstract virtual method which initializes the simple client factory right? LGTM

Yes.

**Comment by bgavrilMS** [comment]
Merging so that it catches 4.46.2 release

**Pull request merged**